### PR TITLE
Develop andrii

### PR DESCRIPTION
**Pull request by AndriiK139MCAF**
Resource Reservation Model

Rajat, Andrii:

We created the class - resource_reservation and added required fields.

We created the tree view, form view and calendar view for resource reservation

We set up the root menu for resource reservation

We created a function inside ResourceReservation class that fetches the current user information and fill "creator" field.

We created the class - resource reservation tag and added required fields

We created the ReservationTag class and added the required fields.

Linkage of resource Tag to Resource Reservation.

Created view for Reservation Tag (tree and form)

We created the class - resource_detail.py and added required fields.

We created the tree view, form view and calendar view for resource reservation

Added view form as button for Resource Detail with functionality of NEW button from Resource Detail (in future NEW will be hide)

Check and refract code using Pylint and Flake

### CODE REVIEW

File: resource_reservation/models/resource_reservation.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        ('completed', 'Completed'),
+        ('other', 'Other'),
+    ],
+        string='Booking Status',
+        default='pending',
+        help="This field represents the status of the booking.")
+    res_des = fields.Text(string="Reservation Description", required=True)
+    # res_tag = fields.Char(string="Reservation Tag", required=True)
+    res_tag_id = fields.Many2one('resource.reservation.tag', string="Resource Tag", required=True)
+
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:47]: Add description for the method

File: resource_reservation/views/menu.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<odoo>
+
+        <!-- Add root menu for  Resource Reservation modul -->
+        <menuitem id="menu_root" name="Resource Reservation"/>
```
Comment (MORPSmcaf) [inline resource_reservation/views/menu.xml:4]: align attributs inside menuitem tags

File: resource_reservation/views/resource_details_views.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+                </tree>
+            </field>
+        </record>
+
+        <!-- Define an action to open the tree view and form view Resource Detail-->
+        <record id="action_resource_detail" model="ir.actions.act_window">
+            <field name="name">Resource Detail</field>
+            <field name="res_model">resource.detail</field>
+            <field name="view_mode">tree,form</field>
+<!--            <field name="create">false</field>-->
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_details_views.xml:23]: remove comments

File: resource_reservation/views/menu.xml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<odoo>
```
Comment (MORPSmcaf) [inline resource_reservation/views/menu.xml:1]: add XML declaration:
<?xml version="1.0" encoding="utf-8"?>

File: resource_reservation/views/resource_details_views.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<odoo>
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_details_views.xml:1]: add XML declaration:
<?xml version="1.0" encoding="utf-8"?>

File: resource_reservation/views/resource_reservation_tag_views.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+<odoo>
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_tag_views.xml:1]: add XML declaration:
<?xml version="1.0" encoding="utf-8"?>

File: resource_reservation/models/resource_reservation.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+class ReservationTag(models.Model):
+    """
+    This class allows admin users to create a reservation tag, edit it, and remove it.
+    This reservation tag has been linked inside the resource reservation class as well.
+    """
+    _name = 'resource.reservation.tag'
+    _description = 'Reservation Tag'
+
+    name = fields.Char(string='Reservation Type', required=True)
+    des = fields.Text(string='Description')
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:14]: rename for clarity

File: resource_reservation/models/resource_reservation.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+
+# Our base model is called resource.reservation
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:17]: Remove word - Our

File: resource_reservation/models/resource_reservation.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        ('confirmed', 'Confirmed'),
+        ('canceled', 'Canceled'),
+        ('completed', 'Completed'),
+        ('other', 'Other'),
+    ],
+        string='Booking Status',
+        default='pending',
+        help="This field represents the status of the booking.")
+    res_des = fields.Text(string="Reservation Description", required=True)
+    # res_tag = fields.Char(string="Reservation Tag", required=True)
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:45]: remove comment

File: resource_reservation/models/resource_reservation.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+        ('pending', 'Pending'),
+        ('confirmed', 'Confirmed'),
+        ('canceled', 'Canceled'),
+        ('completed', 'Completed'),
+        ('other', 'Other'),
+    ],
+        string='Booking Status',
+        default='pending',
+        help="This field represents the status of the booking.")
+    res_des = fields.Text(string="Reservation Description", required=True)
```
Comment (MORPSmcaf) [inline resource_reservation/models/resource_reservation.py:44]: rename for clarity

File: resource_reservation/views/resource_reservation_tag_views.xml [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+                <tree>
+                    <field name="name"/>
+                    <field name="des"/>
+
+                </tree>
+            </field>
+        </record>
+
+
+               <record id="action_resource_reservation_tag" model="ir.actions.act_window">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_tag_views.xml:33]: allign

File: resource_reservation/views/resource_reservation_views.xml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+                            <group>
+                                <field name="name"/>
+                            </group>
+                            <group>
+                                <field name="end_datetime"/>
+
+                            </group>
+                        </group>
+
+                        <!-- Res tag -->
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_views.xml:33]: remove comments

File: resource_reservation/views/resource_reservation_views.xml [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+                        <group>
+
+                            <field name="res_tag_id"/>
+
+
+                        </group>
+
+
+                        <!-- res des and created By -->
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_views.xml:43]: remove comments
align

File: resource_reservation/views/resource_reservation_views.xml [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+                    <field name="end_datetime" width="4"/>
+                    <field name="creator" width="4"/>
+                    <field name="booking_status" width="4"/>
+                </calendar>
+            </field>
+        </record>
+
+
+        <!-- Define an action to open the tree view and form view-->
+        <record id="action_resource_reservation" model="ir.actions.act_window">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_views.xml:95]: id in the end add _act_window

add type <field name="type">ir.actions.act_window</field>

File: resource_reservation/views/resource_reservation_tag_views.xml (hunk 11 above)
Comment (MORPSmcaf) [inline resource_reservation/views/resource_reservation_tag_views.xml:33]: id in the end add _act_window

add type <field name="type">ir.actions.act_window</field>

File: resource_reservation/views/resource_details_views.xml [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+                    <field name="resource_name" width="6"/>
+                    <field name="resource_type" width="6"/>
+                    <field name="resource_capacity" width="6"/>
+                    <field name="resource_owner" width="6"/>
+                </tree>
+            </field>
+        </record>
+
+        <!-- Define an action to open the tree view and form view Resource Detail-->
+        <record id="action_resource_detail" model="ir.actions.act_window">
```
Comment (MORPSmcaf) [inline resource_reservation/views/resource_details_views.xml:19]: id in the end add _act_window

add type <field name="type">ir.actions.act_window</field>

remove comments

File: resource_reservation/views/menu.xml [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+        <menuitem id="menu_resource_reservation" name="Resource Reservation" parent="menu_root"
+                  action="action_resource_reservation"/>
+        <!-- Add resorce detail menu-->
+        <menuitem id="menu_resource_detail" name="Resource Detail" parent="menu_root"
+                  action="action_resource_detail"/>
+
+        <menuitem id="menu_resource_configurations" name="Configurations" parent="menu_root"/>
+        <menuitem id="menu_resource_tag" name="Resource Tags" parent="menu_resource_configurations"
+                  action="action_resource_reservation_tag"/>
```
Comment (MORPSmcaf) [inline resource_reservation/views/menu.xml:14]: _act_window

File: resource_reservation/views/menu.xml [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+         <!-- Add resorce detail menu for create-->
+        <menuitem id="menu_resource_detail_btn" name="Resource Create" parent="menu_resource_configurations"
+                  action="action_resource_detail_btn"/>
```
Comment (MORPSmcaf) [inline resource_reservation/views/menu.xml:17]: _act_window

File: resource_reservation/views/resource_details_views.xml (hunk 3 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_details_views.xml:23]: Its comment for reminding that a NEW button shood be hide 
its removed

File: resource_reservation/views/menu.xml (hunk 2 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/menu.xml:4]: done !

File: resource_reservation/models/resource_reservation.py (hunk 1 above)
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:47]: add comment 
# Fetches information of current user from odoo environment to our model

File: resource_reservation/views/resource_reservation_tag_views.xml (hunk 6 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_reservation_tag_views.xml:1]: done

File: resource_reservation/models/resource_reservation.py (hunk 8 above)
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:17]: done

File: resource_reservation/models/resource_reservation.py (hunk 9 above)
Comment (AndriiK139MCAF) [reply resource_reservation/models/resource_reservation.py:45]: Can we save it, its base field, that can be useful in the future for developing a new feature

File: resource_reservation/views/resource_reservation_views.xml (hunk 12 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_reservation_views.xml:33]: Done

File: resource_reservation/views/menu.xml (hunk 17 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/menu.xml:17]: done

File: resource_reservation/views/menu.xml (hunk 16 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/menu.xml:14]: done

File: resource_reservation/views/resource_details_views.xml (hunk 15 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_details_views.xml:19]: done

File: resource_reservation/views/resource_reservation_tag_views.xml (hunk 11 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_reservation_tag_views.xml:33]: done

File: resource_reservation/views/resource_reservation_views.xml (hunk 14 above)
Comment (AndriiK139MCAF) [reply resource_reservation/views/resource_reservation_views.xml:95]: done

**Pull request closed without merging**